### PR TITLE
Fix coroutine scope canceling log spam (DEV)

### PR DESCRIPTION
**Pull request by d4rken**
:point_up:

### CODE REVIEW

**Review by ralfgehrer — DISMISSED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/flow/HotDataFlow.kt [hunk 1]
```diff
@@ -80,8 +81,11 @@ class HotDataFlow<T : Any>(
             }
         }
         .onCompletion { err ->
-            if (err != null) Timber.tag(tag).w(err, "internal onCompletion due to error")
-            else Timber.tag(tag).v("internal onCompletion")
+            when {
+                err is CancellationException -> Timber.tag(tag).d("internal onCompletion due to cancelation")
+                err != null -> Timber.tag(tag).w(err, "internal onCompletion due to error")
+                else -> Timber.tag(tag).v("internal onCompletion")
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/flow/HotDataFlow.kt:87]: Please include the `err.message` in the log? is not better to use `Timber.e` in this cases?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/BackgroundModeStatus.kt [hunk 2]
```diff
@@ -90,7 +90,10 @@ class BackgroundModeStatus @Inject constructor(
         .onStart { emit(pollisIgnoringBatteryOptimizations()) }
         .distinctUntilChanged()
         .onCompletion {
-            if (it != null) Timber.w(it, "isIgnoringBatteryOptimizations failed.")
+            when {
+                it is CancellationException -> Timber.d("isIgnoringBatteryOptimizations canceled.")
+                it != null -> Timber.w(it, "isIgnoringBatteryOptimizations failed.")
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/BackgroundModeStatus.kt:95]: is `it`  an error ? If so could you please use `Timber.e`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/flow/HotDataFlow.kt (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/flow/HotDataFlow.kt:87]: The signature is `it: Throwable?`

* If it's a cancellation exception we don't care about the specific error message
* If it is an error we log it which would include it's error message automatically ( `.w(err,...)` ) 
* If it is neither of the above, then it was null and we have completion not caused by an exception.
* 
I'll adjust the log level to `e`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/BackgroundModeStatus.kt (hunk 2 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/BackgroundModeStatus.kt:95]: `it: Throwable?`

I'll adjust the log level to `e`

**Review by mtwalli — APPROVED** [review]
thanks

**Review by BMItr — APPROVED** [review]

**Pull request merged**